lib/parallel_tests/test/runner.rb

```diff
@@ -208,15 +208,22 @@ def sort_by_filesize(tests)
           tests.map! { |test| [test, File.stat(test).size] }
         end
 
-        def find_tests(tests, options = {})
-          (tests || []).map do |file_or_folder|
+        def find_tests(tests = [], options = {})
+          files_list = tests.map do |file_or_folder|
             if File.directory?(file_or_folder)
               files = files_in_folder(file_or_folder, options)
-              files.grep(options[:suffix]||test_suffix).grep(options[:pattern]||//)
+              files.grep(options[:suffix] || test_suffix)
+                .grep(options[:pattern] || //)
             else
               file_or_folder
             end
           end.flatten.uniq
+
+          if regex = options[:exclude_pattern]
+            files_list.reject! { |file| file =~ regex }
+          end
```

Comment:
this is not consistent with how pattern works, it should be done inside so it does not apply to folders
... or both should apply to folders 🤷‍♂️ 

... and while your in here: it should use `.flat_map` + `[file_or_folder]` and not `.flatten` and `.uniq!`

Reply:
My assumption would be that anything matching the regex for the include pattern would be included, and the `exclude_pattern` will filter out anything matching the exclude pattern.  It should apply to folders or files, as the exclude pattern behavior should mirror the behavior of native `rspec --exclude-pattern` option.